### ISSUE TITLE
External Images OG open graph tags are coming AFTER the Yoast OG tags

### ISSUE THREAD

**Issue by dhilditch** [superspeedyplugins/issue-tracker#31]
1. Install EI & Yoast
2. Configure Yoast open graph - i.e. enable it
3. Create 2 posts, one with an uploaded featured image and one with an external feature image
4. View source for both - you'll see 2 OG:image tags with ours appearing 2nd

What should happen:

Ideally, we should spot the Yoast OG tags and alter them to reference our external image if an external image exists.

Alternatively, the priority of our action could be moved higher so it runs before Yoast since ours only runs if there is an external image.

**Comment by WebsolsDevelopment** [comment]
@dhilditch after submitting the issue @ discord, we've tried to fix this ourselves for the time being. Maybe you use our code as a starter for your own fix in the plugin:

```
function eiremove_yoastimage_presenter( $presenters ) {
    global $post;
	if(is_single()) {
		global $ei;
		$postid = get_the_ID();
		if ( ! empty( $postid ) && ! empty( $ei ) ) {
			$image_data = $ei->admin->ei_get_image_meta( $postid );
			if ( isset( $image_data['img_url'] ) && $image_data['img_url'] != '' ) {

                return array_map( function ( $presenter ) {
					if ( ! $presenter instanceof Yoast\WP\SEO\Presenters\Open_Graph\Image_Presenter ) {
						return $presenter;
					}
				}, $presenters );

			} else {
				return $presenters;
			}
		}
	}
}
add_action( 'wpseo_frontend_presenters', 'eiremove_yoastimage_presenter' );
```

**Comment by dhilditch** [comment]
Fixed in 3.07
